### ISSUE TITLE
H1 heading in default template

### ISSUE THREAD

**Issue by kirrg001** [conventional-changelog/conventional-changelog#942]
## Current Behavior

We are using the latest Lerna 5.5 version, which generates a CHANGELOG.md based on conventional commits.
But the generated CHANGELOG.md contains multiple H1 headings. I first [raised](https://github.com/lerna/lerna/issues/3324) the issue in Lerna, but they sent me here.

Lerna is using `conventional-changelog-core` [internally](https://github.com/lerna/lerna/blob/main/core/conventional-commits/lib/update-changelog.js).

Heading was already fixed in conventional-changelog:
- https://github.com/conventional-changelog/conventional-changelog/pull/440
- https://github.com/conventional-changelog/conventional-changelog/commit/3d792639815e4c4ae7758e0e84ba72a9cb535f37
- https://github.com/conventional-changelog/conventional-changelog/commit/d2e02d73f275bd10a39c52016999d8faf400c990
- https://github.com/conventional-changelog/conventional-changelog/issues/214
- https://github.com/conventional-changelog/conventional-changelog/issues/867

But it looks like the default template is still using H1:
https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/templates/header.hbs

## Expected Behavior

- Features get H2
- Patches get H3 (or even H2 too)

Thanks!

**Comment by kleinfreund** [comment]
Agreed. This is bugging me as well. Additionally it would be nice if there would always be an h1 at the start of the changelog file (e.g. `# Changelog`).

I just want to point out that the headings for versions should **always** be level 2 headings like so:

```md
# Changelog

## 4.1.1

### Bug fixes

## 4.1.0

### Features
```

Headings for versions should not change in level depending on what kind of release it is because that leads to a broken outline. The following doesn't make sense -- the level 3 heading "4.1.1" structurally follows the level 1 heading "Changelog" so it skips a level without cause.

```md
# Changelog

### 4.1.1

### Bug fixes

## 4.1.0

### Features
```

**Comment by kirrg001** [comment]
@kleinfreund I will look into a fix for that and create a PR asap. Any hints for me?

**Comment by kleinfreund** [comment]
@kirrg001 That’s great! I’m not familiar with this project, but I reckon looking at the following files is going to be pretty relevant:

* https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/templates/header.hbs
* https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/templates/commit.hbs
* https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/templates/footer.hbs

**Comment by kirrg001** [comment]
cc @armano2 Will you help reviewing a PR for this bug?

**Comment by kirrg001** [comment]
cc @bcoe Looking for someone who is actively reviewing pull requests & doing releases for this project.

**Comment by dangreen** [comment]
That's the angular format, not a bug — the preset reproduces Angular's own changelog.

If you want `##` for the version without the level gap, use the `conventionalcommits` preset, it does that out of the box.

To stay on angular, override the header:

```js
import { ConventionalChangelog } from 'conventional-changelog'
import angular from 'conventional-changelog-angular'
import { heading, words, link, compareUrl } from '@conventional-changelog/template'

new ConventionalChangelog()
  .readPackage()
  .config(angular())
  .writer({
    headerPartial: context => heading(2, words(
      context.linkCompare
        ? link(context.version, compareUrl(context))
        : context.version,
      context.date && `(${context.date})`
    ))
  })
```

Changing the default would split existing changelogs: old entries stay `#`, new ones become `##`. Not worth a major on its own, but if there's another breaking pass over the presets, I'll unify the levels there.
